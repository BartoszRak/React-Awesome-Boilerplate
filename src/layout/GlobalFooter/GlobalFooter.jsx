import React from 'react'
import { useCounter } from '~hooks'
import { Button } from '@material-ui/core'

export function GlobalFooter() {
  const { increase, decrease, counted } = useCounter()

  return (
    <div>
      <p>Here is footer with some content:</p>
      <p>
        Actual value:
        {' '}
        {counted}
      </p>
      <Button onClick={increase}>Increase</Button>
      <Button onClick={decrease}>Decrease</Button>
    </div>
  )
}

export default GlobalFooter
