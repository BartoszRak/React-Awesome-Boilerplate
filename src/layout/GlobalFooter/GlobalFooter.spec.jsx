import React from 'react'
import { shallow } from 'enzyme'
import { mocks } from '~utils/tests'

import { GlobalFooter } from './GlobalFooter'

const initComponent = overrides => {
  const mockProps = {
    classes: mocks.classesProxy,
  }
  const mockMethods = {}
  const wrapper = shallow(<GlobalFooter {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('GlobalFooter: ', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
