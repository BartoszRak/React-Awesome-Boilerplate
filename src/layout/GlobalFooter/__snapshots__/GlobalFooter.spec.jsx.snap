// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFooter:  should render as expected 1`] = `
<div>
  <p>
    Here is footer with some content:
  </p>
  <p>
    Actual value:
     
    0
  </p>
  <ButtonMock
    onClick={[Function]}
  >
    Increase
  </ButtonMock>
  <ButtonMock
    onClick={[Function]}
  >
    Decrease
  </ButtonMock>
</div>
`;
